File: practice/src/main/java/com/decade/practice/controllers/validation/ChatIdentifierValidator.java
package com.decade.practice.controllers.validation;

import com.decade.practice.model.domain.embeddable.ChatIdentifier;
import org.springframework.validation.Errors;
import org.springframework.validation.Validator;

public class ChatIdentifierValidator implements Validator {

    @Override
    public boolean supports(Class<?> clazz) {
        return ChatIdentifier.class.isAssignableFrom(clazz);
    }

    @Override
    public void validate(Object target, Errors errors) {
        ChatIdentifier identifier = (ChatIdentifier) target;
        if (identifier.getFirstUser().compareTo(identifier.getSecondUser()) >= 0) {
            errors.reject("Invalid chat identifier");
        }
    }
}

File: practice/src/main/java/com/decade/practice/endpoints/validation/ChatIdentifierValidator.kt
package com.decade.practice.endpoints.validation

import com.decade.practice.model.domain.embeddable.ChatIdentifier
import org.springframework.validation.Errors
import org.springframework.validation.Validator

class ChatIdentifierValidator : Validator {

      override fun supports(clazz: Class<*>): Boolean =
            ChatIdentifier::class.java.isAssignableFrom(clazz)

      override fun validate(target: Any, errors: Errors) {
            val identifier = target as ChatIdentifier
            if (identifier.firstUser >= identifier.secondUser) {
                  errors.reject("Invalid chat identifier")
            }
      }

}